File: backoffice/src/app/features/settings/models/gamificationSettingsDto.ts
/** Réglages de gamification (GET /api/admin/settings/gamification). */
export interface GamificationSettingsDto {
  bronzeThreshold: number;
  silverThreshold: number;
  goldThreshold: number;
  defaultAttendanceReward: number;
}

/** Payload PATCH (bronze est fixe à 0, non modifiable). */
export interface GamificationRequest {
  silverThreshold: number;
  goldThreshold: number;
  defaultAttendanceReward: number;
}

File: api/src/EventHub.Domain/ReadModels/GamificationSettingsDto.cs
namespace EventHub.Domain.ReadModels;

/// <summary>Réglages de gamification exposés au back-office.</summary>
public sealed record GamificationSettingsDto(
    int BronzeThreshold,
    int SilverThreshold,
    int GoldThreshold,
    int DefaultAttendanceReward);
